Revert upstream 0403361ac57082dc47840d1f31832f1a0e319078
It breaks the build when it's defined.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura rsyslog-7.6.6.orig/grammar/lexer.c rsyslog-7.6.6/grammar/lexer.c
--- rsyslog-7.6.6.orig/grammar/lexer.c	2014-10-01 10:12:34.960082719 -0300
+++ rsyslog-7.6.6/grammar/lexer.c	2014-10-01 10:13:24.512769964 -0300
@@ -1459,7 +1459,6 @@
 #line 32 "lexer.l"
 #include "config.h"
 #include "parserif.h"
-extern char *strdup(char*); /* somehow we do not get this from string.h... */
 /*%option noyywrap nodefault case-insensitive */
 /* avoid compiler warning: `yyunput' defined but not used */
 #define YY_NO_INPUT 1
diff -Nura rsyslog-7.6.6.orig/grammar/lexer.l rsyslog-7.6.6/grammar/lexer.l
--- rsyslog-7.6.6.orig/grammar/lexer.l	2014-10-01 10:12:34.960082719 -0300
+++ rsyslog-7.6.6/grammar/lexer.l	2014-10-01 10:13:41.935363172 -0300
@@ -31,7 +31,6 @@
 %{
 #include "config.h"
 #include "parserif.h"
-extern char *strdup(char*); /* somehow we do not get this from string.h... */
 %}
 
 %option noyywrap nodefault case-insensitive yylineno
